### PR TITLE
Update all components

### PR DESCRIPTION
**Pull request by nshahan**
* Update dependency to angular: ^4.0.0.
 * App Layout: Fix margins for header icons/links.
 * Material Button: Refactor raised styling so it can be used in a mixin.
 * Material Input: Add support for material dark theme.
 * Material Popup:
   * Constrain popups to the viewport if both `enforceSpaceConstraints` and
     `overlayRepositionLoop` are enabled.
   * Create the material-popup view synchronously.
   * Remove `animationComplete` event.
 * Add missing `MaterialIconComponent`, `MaterialPersistentDrawerDirective`,
   `MaterialTemporaryDrawerDirective`, and `MaterialSelectSearchboxComponent` to
   `materialDirectives`.
 * Change DisplayNameDirective to a functional directive.
 * Additional styling cleanup to support `ng-deep`.
 * Reduce visibility of common directives/components for smaller code size and
   better performance.
 * Refactor Angular annotations to inline versions.
 * Update documentation.

### CODE REVIEW

File: lib/angular_components.dart [hunk 1]
```diff
@@ -11,7 +11,7 @@ import 'src/all_components.dart';
 export 'src/all_components.dart';
 
 /// A convenience list of all Directives exposed by this package.
-const List<Type> materialDirectives = const [
+const List<dynamic> materialDirectives = const [
```
Comment (kevmoo) [inline lib/angular_components.dart]: What in here isn't a `Type`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,24 @@
+## 0.6.0
+
+ * Update dependency to angular: ^4.0.0.
```
Comment (kevmoo) [inline CHANGELOG.md]: Super fun, I know – would you collapse the changelog into one entry for v0.6.0?

File: lib/angular_components.dart (hunk 1 above)
Comment (nshahan) [reply lib/angular_components.dart]: DisplayNameRenderDirective changed in this update. It is now a "functional directive".
https://github.com/dart-lang/angular_components/pull/153/files#diff-a8788133997a1100324e3684b3b3e6c3

File: CHANGELOG.md (hunk 2 above)
Comment (nshahan) [reply CHANGELOG.md]: Oh yeah, I was wondering about that. I'll go through and smash them together.

File: lib/angular_components.dart (hunk 1 above)
Comment (kevmoo) [reply lib/angular_components.dart]: Ah! Cool...

File: CHANGELOG.md (hunk 2 above)
Comment (kevmoo) [reply CHANGELOG.md]: 👍

**Pull request merged**